utils/s2n_init.c

```diff
@@ -52,6 +48,7 @@ int s2n_disable_atexit(void) {
 int s2n_init(void)
 {
     main_thread = pthread_self();
+    POSIX_GUARD_RESULT(s2n_libcrypto_validate_runtime());
```

Comment:
and this doesn't need to run after `s2n_fips_init` because `s2n_libcrypto_is_fips` now just calls libcrypto instead of returning a variable ?

Reply:
`s2n_libcrypto_validate_runtime()` doesn't validate the fips state of the libcrypto library. That is handled by `s2n_fips_init()` - which cannot fail. The name does not seem to align with its semantics(?).

A consumer cannot (currently) build a libs2n (against fips openssl) and then assert s2n it is in fips mode at run-time.